data/shaders/planet.frag

```diff
@@ -26,6 +26,7 @@ varying highp vec3 P; //original vertex pos in model space
 
 uniform sampler2D tex;
 uniform mediump vec2 poleLat; //latitudes of pole caps, in terms of texture coordinate. x>0...north, y<1...south. 
+uniform bool renderPolarCaps;

```

Comment:
I hope to solve this by correctly setting poleLat for 3D objects. Let me check that first please.